prog/vm/nexus.rb

```diff
@@ -152,6 +152,7 @@ def before_run
       if strand.label != "destroy"
         vm.active_billing_records.each(&:finalize)
         vm.assigned_vm_address&.active_billing_record&.finalize
+        vm.update(has_customer_data: true)
```

Comment:
Well, this is a little bit weird to mark it at the destroy time. Why is this necessary?

Reply:
We don't want to pick a vm which is already deleting

Reply:
Yeah, it looks weird changed it from tainted to has_customer_data